src/seeds.history.cpp

```diff
@@ -354,6 +360,85 @@ void history::save_from_metrics (name from, int64_t & from_points, int64_t & qua
   }
 }
 
+void history::send_trx_cbp_reward_action (name from, name to) {
+  action a(
+    permission_level(get_self(), "active"_n),
+    get_self(),
+    "sendtrxcbp"_n,
+    std::make_tuple(from, to)
+  );
+
+  transaction tx;
+  tx.actions.emplace_back(a);
+  tx.delay_sec = 1;
+  tx.send(from.value + 10, _self);
+}
+
+void history::send_add_cbs (name account, int points) {
+  action(
+    permission_level(contracts::accounts, "active"_n),
+    contracts::accounts,
+    "addcbs"_n,
+    std::make_tuple(account, points)
+  ).send();
+}
+
+void history::trx_cbp_reward (name account, name key) {
+
+  auto trxcbprewards_by_acct_key = trxcbprewards.get_index<"byacctkey"_n>();
+
+  uint128_t id = (uint128_t(account.value) << 64) + key.value;
+  auto itr = trxcbprewards_by_acct_key.find(id);
+
+  uint64_t now = eosio::current_time_point().sec_since_epoch();
+
+  if (itr != trxcbprewards_by_acct_key.end()) {
+
+    uint64_t threshold = now - utils::moon_cycle;
+    if (itr->timestamp > threshold) { return; }
```

Comment:
So I guess this encodes that each reward can only be paid out 1x per moon cycle?

Makes sense...